### PR TITLE
30 refine configuration

### PR DESCRIPTION
**Pull request by KappaDistributive**
solves #30 

This should only be reviewed once #28 has been merged.

### CODE REVIEW

File: model/config/config.yaml [hunk 1]
```diff
@@ -13,10 +13,50 @@ logger:
...
-        insta-caption-train.json: train
-        insta-caption-test1.json: eval
-        insta-caption-test2.json: test 
+        insta-caption-train.json: "train"
+        insta-caption-test1.json: "eval"
+        insta-caption-test2.json: "test"
+
+Trainer:
+    pad_token_id: 0
+    unk_token_id: 0
```
Comment (riven314) [inline model/config/config.yaml:26]: why UNK and PAD has the same index?
Also, when I look back the pretrained word embedding layer in `AttentionLSTMDecoder`, it seems there is no enforcement as to align these index to the pretrained word embedding's. 
(e.g `eos_token_id` is never used in `AttentionLSTMDecoder`, `bos_token_id` is not applied to the pretrained embedding)

File: model/config/config.yaml [hunk 2]
```diff
@@ -13,10 +13,50 @@ logger:
...
+    ShowAttendTell:
+        ResnetEncoder:
+            input_height: 320
+            input_width: 320
+            hidden_dim: 300
+
+        AttentionLSTMDecoder:
+            hidden_dimension: 300
+            embedding_dimension: 300
+            vocabulary_size: 100000
```
Comment (riven314) [inline model/config/config.yaml:56]: is this the minimum vocab size of our IG dataset?

File: model/config/config.yaml [hunk 3]
```diff
@@ -13,10 +13,50 @@ logger:
...
+            encoder_output_dimension: 2048
+            number_of_lstm_layers: 1
+            unk_token_id: 0
```
Comment (riven314) [inline model/config/config.yaml:59]: why these token-id-related spec repeated twice?

File: model/config/config.yaml (hunk 1 above)
Comment (KappaDistributive) [reply model/config/config.yaml:26]: > why UNK and PAD has the same index?

This pretrained embedding doesn't come with a <pad> token, so I'm using its <unk> token instead. That's fine because I ignore it in the loss function.

> it seems there is no enforcement

That's taken care of in the preprocessor. See https://github.com/riven314/TweakStory/blob/8ed4cb01125d239e8a534fecd9c146136cf7835d/model/preprocessor.py#L482

File: model/config/config.yaml (hunk 2 above)
Comment (KappaDistributive) [reply model/config/config.yaml:56]: It's the vocabulary size of the pretrained embedding. We'll likely use a different value here. But determining the best choice is left for another PR.

File: model/config/config.yaml (hunk 3 above)
Comment (KappaDistributive) [reply model/config/config.yaml:59]: Because of the way I currently deal with configs. It's not ideal... and there are ways to get rid of these duplicates.

Probably the easiest way would be to split out these kinds of settings and `!include` them where needed. What do you think?

Comment (riven314) [reply model/config/config.yaml:59]: I prefer leaving it here if only `unk_token_id` is duplicated.
